pyk/src/pyk/integration_tests/test_emit_json_spec.py

```diff
@@ -1,35 +1,38 @@
 import json
 from pathlib import Path
 
-from kprove_test import KProveTest
-
-from pyk.kast import (
-    EMPTY_ATT,
-    KApply,
-    KAst,
-    KClaim,
-    KDefinition,
-    KRequire,
-    KVariable,
-)
-from pyk.kastManip import rewriteAnywhereWith
+from ..kast import EMPTY_ATT, KAst, KDefinition, KRequire
+from ..kastManip import remove_generated_cells
+from ..ktool import KompileBackend, kprovex
+from .kprove_test import KProveTest
 
 
 class EmitJsonSpecTest(KProveTest):
-    DEFN_DIR = 'definitions/imp-verification/haskell/imp-verification-kompiled'
     MAIN_FILE_NAME = 'imp-verification.k'
-    USE_DIR = '.emit-json-spec-test'
-    INCLUDE_DIRS = ['k-files']
-    SPEC_FILE = 'specs/imp-verification/looping-spec.json'
+
+    KOMPILE_MAIN_FILE = f'k-files/{MAIN_FILE_NAME}'
+    KOMPILE_BACKEND = KompileBackend.HASKELL
+    KOMPILE_OUTPUT_DIR = 'definitions/imp-verification/haskell'
+    KOMPILE_EMIT_JSON = True
+
+    KPROVE_USE_DIR = '.emit-json-spec-test'
+
+    SPEC_FILE = 'k-files/looping-spec.k'
+    SPEC_JSON_FILE = 'definitions/imp-verification/haskell/looping-spec.json'
 
     def setUp(self):
         super().setUp()
 
-        with open(self.SPEC_FILE, 'r') as spec_file:
+        spec_file = Path(self.SPEC_FILE)
+        directory = Path(self.KOMPILE_OUTPUT_DIR)
+        emit_json_spec = Path(self.SPEC_JSON_FILE)
+        kprovex(spec_file, directory=directory, emit_json_spec=emit_json_spec, dry_run=True)
```

Comment:
Can this use `self.kprove` instead? Or should it be calling the `kprovex` function directly? If `KProve` is made to call `kprovex` function, I guess this should go through the class interface.

Reply:
Tried with `self.kprove.prove`, but it passes spurious flags to `kprove` (e.g. `--spec-module`).